internal/provider/services/service.go

```diff
@@ -0,0 +1,21 @@
+package services
+
+import (
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type Service interface {
+	RegisterWithServer(grpcServer grpc.ServiceRegistrar) error
```

Comment:
As mentioned in another comment, lets move this definition up to a package (create one if needs be) common to both Module and Provider

Reply:
yes, created a new package for gateway and service which is now common to both modules & providers.